src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -4822,6 +4822,184 @@ .ForwardHelpCategory Cmdlet
 ";
         }
 
+#if UNIX
+        internal static string GetSudoFunctionText()
+        {
+            return @"
+Set-StrictMode -Version Latest
+
+$script:AmbiguousOperators = @('>', '<', '|', '&')
+$script:IgnoredTokens = @(
+    [System.Management.Automation.PSTokenType]::CommandParameter,
+    [System.Management.Automation.PSTokenType]::CommandArgument,
+    [System.Management.Automation.PSTokenType]::NewLine,
+    [System.Management.Automation.PSTokenType]::Comment,
+    [System.Management.Automation.PSTokenType]::StatementSeparator,
+    [System.Management.Automation.PSTokenType]::LineContinuation
+    )
+
+function Get-SudoArgs
+{
+    param
+    (
+        [System.Collections.Generic.List[string]] $AllArgs
+    )
+    $sb = [System.Text.StringBuilder]::new()
+
+    [string] $arg = $AllArgs[0]
+    while ($arg.StartsWith('-'))
+    {
+        $AllArgs.RemoveAt(0)
+        $null = $sb.Append(' ')
+        $null = $sb.Append($arg)
+
+        if ($AllArgs.Count -gt 0)
+        {
+            $arg = $AllArgs[0]
+        }
+        else
+        {
+            break
+        }
+    }
+    return $sb.ToString()
+}
+
+function Get-ScriptBlock
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [System.Collections.Generic.List[string]] $AllArgs,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string] $SudoCmd
+    )
+    [bool] $usePowerShell = $false
+
+    [string] $sudoArgs = Get-SudoArgs -AllArgs $AllArgs
+
+    <# tokenize the remaining arguments to identify the type of expression; native or powershell
```

Comment:
Right, I considered that approach but found it a little cumbersome to use.  I can impose it but it means it will be error prone for the natural use case:

sudo get-item /etc/bin/foo
versus
sudo {get-item  /etc/bin/foo}
I'll revisit that path again and see how it impacts extracting sudo-specific arguments.